Orange/widgets/regression/owunivariateregression.py

```diff
@@ -186,20 +182,19 @@ def plot_regression_line(self, x_data, y_data):
         self.plotview.replot()
 
     def apply(self):
-        learner = self.learner
         predictor = None
 
-        if self.data is not None:
-
-            degree = int(self.polynomialexpansion)
-            learner = self.LEARNER(preprocessors=self.preprocessors,
-                                   degree=degree,
-                                   learner=LinearRegressionLearner() if self.learner is None
-                                    else learner)
+        degree = int(self.polynomialexpansion)
+        learner = self.LEARNER(
+            preprocessors=self.preprocessors,
+            degree=degree,
+            learner=LinearRegressionLearner() if self.learner is None else self.learner)
 
+        if self.data is not None:
             attributes = self.x_var_model[self.x_var_index]
             class_var = self.y_var_model[self.y_var_index]
-            data_table = Table(Domain([attributes], class_vars=[class_var]), self.data)
+            data_table = Table(Domain([attributes], class_vars=[class_var]),
+                               self.data)
 
             learner.name = self.learner_name
```

Comment:
Doesn't this belong before `if`, too? The name should be set even if there's no data.